include/flash.h

```diff
@@ -127,6 +137,28 @@ static inline int flash_write_protection_set(struct device *dev, bool enable)
 	return api->write_protection(dev, enable);
 }
 
+/**
+ *  @brief  Get layout of pages of flash.
```

Comment:
I am saying that:

- flash_page_layout() should not be part of the include/flash.h public API
- page_layout should remain a part of struct flash_driver_api
- some generic flash routines for finding out sector boundaries given an address should be written, which use the run length encoding in flash_driver_api.page_layout to do the computation, since it's always the same for every flash part. Only the layout changes, not the algorithm.